kivy/graphics/vertex_instructions_line.pxi

```diff
@@ -1154,47 +1209,100 @@ cdef class Line(VertexInstruction):
+        #  adjustments.
+        if c1 > half_min_dimension:
+            c2 = min(c2, half_min_dimension)
+            c4 = min(c4, half_min_dimension)
+            c1 = min(c1, min_dimension - c2, min_dimension - c4)
+
+        if c2 > half_min_dimension:
+            c1 = min(c1, half_min_dimension)
+            c3 = min(c3, half_min_dimension)
+            c2 = min(c2, min_dimension - c1, min_dimension - c3)
+        
+        if c3 > half_min_dimension:
+            c2 = min(c2, half_min_dimension)
+            c4 = min(c4, half_min_dimension)
+            c3 = min(c3, min_dimension - c2, min_dimension - c4)
+        
+        if c4 > half_min_dimension:
+            c3 = min(c3, half_min_dimension)
+            c1 = min(c1, half_min_dimension)
+            c4 = min(c4, min_dimension - c3, min_dimension - c1)
+
+        # Resulting rounded_rectangle
+        self._rounded_rectangle = (x, y, w, h, c1, c2, c3, c4, resolution)
+        # Reset another properties
+        self._rectangle = self._ellipse = self._circle = None
+
+        step = PI / resolution
+        max_a = PI / 2.0 - step
+        
+        # top-left
+        a = 0.0
         px = x + c1
-        py = y + c1
+        py = y + h - c1
 
-        while a < - PI / 2.:
-            a += pi / resolution
+        while a < max_a:
+            a += step
             self._points.extend([
-                px + cos(a) * c1,
-                py + sin(a) * c1])
+                px - cos(a) * c1,
+                py + sin(a) * c1
+            ])
 
+        # top-right
+        a = 0.0
         px = x + w - c2
-        py = y + c2
+        py = y + h - c2
 
-        while a < 0:
-            a += PI / resolution
+        while a < max_a:
+            a += step
             self._points.extend([
-                px + cos(a) * c2,
-                py + sin(a) * c2])
+                px + sin(a) * c2,
+                py + cos(a) * c2
+            ])
 
+        # bottom-right
+        a = 0.0
         px = x + w - c3
-        py = y + h - c3
+        py = y + c3
 
-        while a < PI / 2.:
-            a += PI / resolution
+        while a < max_a:
+            a += step
             self._points.extend([
                 px + cos(a) * c3,
-                py + sin(a) * c3])
+                py - sin(a) * c3
+            ])
 
+        # bottom-left
+        a = 0.0
         px = x + c4
-        py = y + h - c4
+        py = y + c4
 
-        while a < PI:
-            a += PI / resolution
+        while a < max_a:
+            a += step
             self._points.extend([
-                px + cos(a) * c4,
-                py + sin(a) * c4])
+                px - sin(a) * c4,
+                py - cos(a) * c4
+            ])
 
         self._close = 1
 
     property bezier:
         '''Use this property to build a bezier line, without calculating the
-        :attr:`points`. You can only set this property, not get it.
+        :attr:`points`.
```

Comment:
did you also mean to save `_bezier` and to reset the others in this setter? i don't see it.

Reply:
It was also an unintentional change. Thanks for identifying!